thorclient/httpclient/http_request.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package httpclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/vechain/thor/v2/thorclient/common"
+)
+
+func (c *Client) httpRequest(method, url string, payload io.Reader) ([]byte, error) {
+	body, statusCode, err := c.rawHTTPRequest(method, url, payload)
+	if err != nil {
+		return nil, err
+	}
+	if statusCode != http.StatusOK {
+		return nil, fmt.Errorf("http error - Status Code %d - %s - %w", statusCode, body, common.ErrNot200Status)
+	}
+	return body, nil
+}
+
+func (c *Client) rawHTTPRequest(method, url string, payload io.Reader) ([]byte, int, error) {
+	req, err := http.NewRequest(method, url, payload)
+	if err != nil {
+		return nil, 0, fmt.Errorf("error creating request: %w", err)
+	}
+
+	if method == "POST" {
+		req.Header.Set("Content-Type", "application/json")
+		req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
```

Comment:
Do we need `application/x-www-form-urlencoded`?

Reply:
Fixed 👍